### PR TITLE
Basic Tours Search

### PR DESCRIPTION
**Pull request by tpbnick**
# Overview
Implements a basic search feature for tours in the back-end.  This search looks through tour names and descriptions.

# Testing
I recommend creating a few tours before testing so you can see how the filtering works.  I use [Postman ](https://www.postman.com/downloads/)to create tours currently.

This is my setup to create tours in Postman:
image
The POST URL will most likely be different on your device.  Check the URL that the Flask server gives you when it starts running and use that:
image

1. Start up the flask server
2. Open the Flutter App
3. Choose the Search option from the Hamburger Menu
4. Try entering some text

image

Eventually, we will need to make clicking on a list item take the user to the correct tour.

# Comments
There has to be some data returned from the `/api/tours` endpoint for searching to be possible.  If there are no tours returns (and the status code is 200), there will be a popup saying "There are currently no tours to search for."  We can add another button in the future that lets the user go to the Create Tour page if we want.

### CODE REVIEW

**Comment by nguyendviet** [comment]
@tpbnick I know this feature works. Can you resolve the conflict? We can merge it.

**Comment by tpbnick** [comment]
@nguyendviet should be good now.  Double check my changes to the `hamburger.dart` file though, I am not sure if that merge was perfect.

File: virotour/lib/src/search.dart [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                              builder: (context) => const TourDetailsView(),
+                            ),
+                          );
+                        },
+                        child: ElevatedButton(
+                          onPressed: () {
+                            Navigator.push(
+                              context,
+                              MaterialPageRoute(
+                                builder: (context) => const TourDetailsView(),
```
Comment (nguyendviet) [inline virotour/lib/src/search.dart]:
```suggestion
                                builder: (context) => const TourDetailsView(), // Tour now requires the tour info
```
I think you'll need to update your code with the tours found from your search and pass the tour's details into the class `Tour`.
Screenshot 2023-03-15 at 11 12 25

**Review by nguyendviet — CHANGES_REQUESTED** [review]

File: virotour/lib/src/search.dart (hunk 1 above)
Comment (tpbnick) [reply virotour/lib/src/search.dart]: Fixed.  I also needed to update the `searchData()` function.

**Comment by tpbnick** [comment]
@nguyendviet ready for re-review.

**Review by nguyendviet — APPROVED** [review]
Excellent work! @tpbnick

**Pull request merged**